tests/UnitTests/PhasorDynamics/GenrouTests.hpp

```diff
@@ -45,7 +45,43 @@ namespace GridKit
       TestOutcome residual()
       {
         TestStatus success = true;
-        success.skipTest();
+
+        PhasorDynamics::Bus<ScalarT, IdxT>    bus(1.0, 0.0);
+        PhasorDynamics::Genrou<ScalarT, IdxT> gen(&bus,
+                                                  1,
+                                                  1,
+                                                  0.05013,
+                                                  3,
+                                                  0,
+                                                  0,
+                                                  7,
+                                                  0.04,
+                                                  0.05,
+                                                  0.75,
+                                                  2.1,
+                                                  0.2,
+                                                  0.18,
+                                                  0.5,
+                                                  0.5,
+                                                  0.18,
+                                                  0.15,
+                                                  0,
+                                                  0);
+
+        bus.allocate();
+        bus.initialize();
+        bus.evaluateResidual();
+
+        gen.allocate();
+        gen.initialize();
+        gen.evaluateResidual();
+
+        const std::vector<ScalarT>& f = gen.getResidual();
+        for (IdxT i = 0; i < 21; ++i)
+        {
+          if (!isEqual(f[i], 0.0, 1e-14))
+            success = false;
+        }
```

Comment:
```suggestion
        for (auto f_val : f)
        {
          if (!isEqual(f_val, 0.0, 1e-14))
            success = false;
        }
```

Best convert this to a range-based for in case the dimension of the residual increases later and we accidentally miss a value not matching because we forgot to increase the bounds.

Reply:
Fixed in [828e780](https://github.com/ORNL/GridKit/pull/85/commits/828e7803d2e348b94cdcd2c1ac8f050e64697775).